substrate/frame/core-fellowship/src/migration.rs

```diff
@@ -68,34 +68,141 @@ mod v0 {
+		}
+
+		fn on_runtime_upgrade() -> frame_support::weights::Weight {
+			// Read the old value from storage
+			let old_value = v0::Params::<T, I>::take();
+			// Write the new value to storage
+			let new = ParamsOf::<T, I> {
+				active_salary: BoundedVec::defensive_truncate_from(
+					old_value.active_salary.to_vec(),
+				),
+				passive_salary: BoundedVec::defensive_truncate_from(
+					old_value.passive_salary.to_vec(),
+				),
+				demotion_period: BoundedVec::defensive_truncate_from(
+					old_value.demotion_period.to_vec(),
+				),
+				min_promotion_period: BoundedVec::defensive_truncate_from(
+					old_value.min_promotion_period.to_vec(),
+				),
+				offboard_timeout: old_value.offboard_timeout,
+			};
+			Params::<T, I>::put(new);
+			T::DbWeight::get().reads_writes(1, 1)
+		}
+	}
+}
+
+pub mod v2 {
+	use super::*;
+	use crate::BlockNumberFor as NewBlockNumberFor;
+	use frame_system::pallet_prelude::BlockNumberFor as LocalBlockNumberFor;
+
+	/// Converts previous (local) block number into the new one. May just be identity functions
+	/// if sticking with local block number as the provider.
+	pub trait ConvertBlockNumber<L, N> {
+		/// Converts to the new type and finds the equivalent moment in time as relative to the new
+		/// block provider
+		///
+		/// For instance - if your new version uses the relay chain number, you'll want to
+		/// use relay current - ((current local - local) * equivalent_block_duration)
+		fn equivalent_moment_in_time(local: L) -> N;
+
+		/// Returns the equivalent time duration as the previous type when represented as the new
+		/// type
+		///
+		/// For instance - If you previously had 12s blocks and are now following the relay chain's
+		/// 6, one local block is equivalent to 2 relay blocks in duration
+		fn equivalent_block_duration(local: L) -> N;
 	}
 
-	fn on_runtime_upgrade() -> frame_support::weights::Weight {
-		// Read the old value from storage
-		let old_value = v0::Params::<T, I>::take();
-		// Write the new value to storage
-		let new = crate::ParamsType {
-			active_salary: BoundedVec::defensive_truncate_from(old_value.active_salary.to_vec()),
-			passive_salary: BoundedVec::defensive_truncate_from(old_value.passive_salary.to_vec()),
-			demotion_period: BoundedVec::defensive_truncate_from(
-				old_value.demotion_period.to_vec(),
-			),
-			min_promotion_period: BoundedVec::defensive_truncate_from(
-				old_value.min_promotion_period.to_vec(),
-			),
-			offboard_timeout: old_value.offboard_timeout,
-		};
-		crate::Params::<T, I>::put(new);
-		T::DbWeight::get().reads_writes(1, 1)
+	pub struct MigrateToV2<T, BlockNumberConverter, I = ()>(
+		PhantomData<(T, BlockNumberConverter, I)>,
+	);
+
+	impl<T: Config<I>, BlockNumberConverter, I: 'static> UncheckedOnRuntimeUpgrade
+		for MigrateToV2<T, BlockNumberConverter, I>
+	where
+		BlockNumberConverter: ConvertBlockNumber<LocalBlockNumberFor<T>, NewBlockNumberFor<T, I>>,
+	{
+		fn on_runtime_upgrade() -> frame_support::weights::Weight {
```

Comment:
we could count the number of member in `pre_upgrade` and then ensure the number of member is same in `post_upgrade`
And same query if there is a value in params in `pre_upgrade` and then ensure there is a value in params in `post_upgrade`.